src/main/java/graphql/util/FpKit.java

```diff
@@ -279,13 +280,13 @@ public static OptionalInt toSize(Object iterableResult) {
     }
 
     public static <T> Set<T> filterSet(Collection<T> input, Predicate<T> filter) {
-        LinkedHashSet<T> result = new LinkedHashSet<>();
+        ImmutableSet.Builder<T> result = ImmutableSet.builder();
         for (T t : input) {
             if (filter.test(t)) {
                 result.add(t);
             }
         }
-        return result;
+        return result.build();
     }
```

Comment:
This is used in 2 places (overlapping fields code as well) and both work as expected